astroquery/esa/iso/core.py

```diff
@@ -0,0 +1,328 @@
+        link = link + "&DATA_RETRIEVAL_ORIGIN=astroquery"
+        link = link + "&tdt=" + tdt
+
+        if verbose:
+            log.info(link)
+
+        return link
+
+    def get_postcard(self, tdt, *, filename=None, verbose=False):
+        """
+        Download postcards from ISO Data Archive
+
+        Parameters
+        ----------
+        tdt : string
+            id of the observation for which download the postcard, mandatory
+            The identifier of the observation we want to retrieve, regardless
+            of whether it is simple or composite.
+        filename : string
+            file name to be used to store the postcard, optional, default None
+        verbose : bool
+            optional, default 'False'
+            Flag to display information about the process
+
+        Returns
+        -------
+        None. It downloads the observation postcard indicated
+        """
+
+        link = self.get_postcard_link(tdt, filename, verbose)
+
+        local_filepath = self._request('GET', link, cache=True, save=True)
+
+        if filename is None:
+            try:
+                response = self._request('HEAD', link)
+                response.raise_for_status()
+            except requests.exceptions.RequestException as e:
+                print(str(e))
+                raise SystemExit(e)
+
+            filename = re.findall('filename="(.+)"', response.headers[
+                "Content-Disposition"])[0]
+        else:
+            filename = filename + ".png"
+
+        log.info("Copying file to {0}...".format(filename))
+
+        shutil.move(local_filepath, filename)
+
+        if verbose:
+            log.info("Wrote {0} to {1}".format(link, filename))
+
+        return filename
+
+    def query_ida_tap(self, query, *, output_file=None,
+                      output_format="votable", verbose=False):
+        """
+        Launches a synchronous job to query the ISO Tabular Access Protocol Service
+
+        Parameters
+        ----------
+        query : str, mandatory
+            query (adql) to be executed
+        output_file : str, optional, default None
+            file name where the results are saved if dumpToFile is True.
+            If this parameter is not provided, the jobid is used instead
+        output_format : str, optional, default 'votable'
+            possible values 'votable' or 'csv'
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A table object
+        """
+
+        job = self._tap.launch_job(query=query, output_file=output_file,
+                                   output_format=output_format,
+                                   verbose=verbose,
+                                   dump_to_file=output_file is not None)
+        try:
+            table = job.get_results()
+            return table
+        except HTTPError:
+            print(table)
```

Comment:
this block is not valid: if `job.get_results()` raises an `HTTPError`, `table` will not be defined.

Reply:
print exception instead of possible not defined table